### PR TITLE
feat(radio-button): adds animation

### LINKED ISSUE #10609: [Radio Button]: Micro-interactions while hovering and selecting
### Summary

This issue enhances Carbon's native Radio Button component. Currently, the Radio button component does not show any micro-interactions especially when you are hovering over the element and selecting another radio button. 

Motion designer:
Tyler Watson
Slack: tyler.watson

Demo:
radioButton_light

UX Specs:
image (1)

### Justification

Radio buttons are a great place to add micro-interactions especially when users are repeatedly using radio buttons in forms. The animation adds a breath of fresh air and makes the process a little fun, and exciting. 

### Desired UX and success metrics

Users would be able to select a radio button and see a micro-interaction when hovering over other options in the radio button group. 

### Required functionality

_No response_

### Specific timeline issues / requests

_No response_

### Available extra resources

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by AlexanderMelox**
Closes #10609 

This pull request adds micro-interactions to the RadioButton component.

https://user-images.githubusercontent.com/12755042/157718499-05de4ae9-cc40-4230-aaf1-a7861f611dac.mov

#### Changelog

**New**

- Adds new hovers and active states to the radio button

**Changed**

-  packages/components/src/components/radio-button/_radio-button.scss
-  packages/styles/scss/components/radio-button/_radio-button.scss

#### Testing / Reviewing

- Used storybook to make sure changes do not change visuals and all props and controls are still functional.
- Made sure animations are still keyboard accessible.

### CODE REVIEW

**Comment by mjabbink** [comment]
@AlexanderMelox I like it. Should tabbing through the radio buttons work on this preview?

**Comment by AlexanderMelox** [comment]
Yes it should work! I double checked

**Review by joshblack — COMMENTED** [review]
Thanks for taking the time to do this plus port it over to the styles package! Really appreciate it 🙏  

I think the only thing I noticed is that the animation seems to still apply for the disabled state and I wanted to check if this was intended.

https://user-images.githubusercontent.com/3901764/157699961-cc731894-5ee7-4839-8520-2b92066c6452.mov

File: packages/components/src/components/radio-button/_radio-button.scss [hunk 1]
```diff
@@ -136,6 +148,7 @@
     .#{$prefix}--radio-button__appearance {
     outline: 2px solid $focus;
     outline-offset: 1.5px;
```
Comment (tw15egan) [inline packages/components/src/components/radio-button/_radio-button.scss]: Not included in your changes, but while reviewing I noticed that Firefox is calculating this incorrectly and causing a weird focus. 

image

Can we just change this to 1px?

File: packages/styles/scss/components/radio-button/_radio-button.scss [hunk 2]
```diff
@@ -147,6 +160,7 @@ $radio-border-width: 1px !default;
     .#{$prefix}--radio-button__appearance {
     outline: 2px solid $focus;
     outline-offset: 1.5px;
```
Comment (tw15egan) [inline packages/styles/scss/components/radio-button/_radio-button.scss]: Same as above

File: packages/styles/scss/components/radio-button/_radio-button.scss [hunk 3]
```diff
@@ -106,7 +102,9 @@ $radio-border-width: 1px !default;
       background-color: $icon-primary;
       border-radius: 50%;
       content: '';
-      transform: scale(0.5);
+      transform: scale(0);
+      transition: background-color $duration-fast-02 motion(exit, productive),
+        transform $duration-moderate-01 motion(exit, productive);
```
Comment (tw15egan) [inline packages/styles/scss/components/radio-button/_radio-button.scss:111]: We recently added an a11y helper to disable animations when `prefers-reduced-motion` is enabled. Can we add this media query to this style block?

```css
      @media (prefers-reduced-motion: reduce) {
        transition: none;
      }
```

File: packages/styles/scss/components/radio-button/_radio-button.scss [hunk 4]
```diff
@@ -198,4 +212,33 @@ $radio-border-width: 1px !default;
...
+    .#{$prefix}--radio-button__appearance {
+    border-color: $toggle-off;
+    transition: border-color $duration-fast-02 motion(entrance, productive);
+
+    &::before {
+      background-color: $icon-secondary;
+      transform: scale(0.5);
+      transition: background-color $duration-fast-02
+          motion(entrance, productive),
+        transform $duration-moderate-01 motion(entrance, productive);
```
Comment (tw15egan) [inline packages/styles/scss/components/radio-button/_radio-button.scss:230]: Same comment as above regarding reduced motion.

**Review by tw15egan — CHANGES_REQUESTED** [review]
The animation looks crisp! Just had a few comments

**Comment by AlexanderMelox** [comment]
> Thanks for taking the time to do this plus port it over to the styles package! Really appreciate it 🙏
> 
> I think the only thing I noticed is that the animation seems to still apply for the disabled state and I wanted to check if this was intended.
> 
>  Screen.Recording.2022-03-10.at.09.51.36.mov

@joshblack Interesting in the development storybook it wasn't applying to the disabled state 🤔 I'll take a look at it, thanks!

**Comment by kingtraceyj** [comment]
From design: The hover states on the text work differently than Carbon's current paradigm. I suggest simplifying the animation to just the animation and not including the color changes on hover.

**Comment by AlexanderMelox** [comment]
I will be out of office for a week, but will get back to these feedback next week.

**Comment by watsondesign** [comment]
@kingtraceyj there appears to be components in Carbon where the type changes on hover (for example the tabs). My intention with the color change is to give the user feedback into which option is on hover. Without it, I don't believe it too be as effective

**Comment by AlexanderMelox** [comment]
@kingtraceyj @watsondesign I went ahead and removed the hover animation on the text. Can we get this reviewed so we can merge this PR?

**Comment by AlexanderMelox** [comment]
@kingtraceyj @laurenmrice Can we get this PR reviewed?

**Comment by kingtraceyj** [comment]
@AlexanderMelox and @watsondesign would you mind sharing this at our design office hours perhaps in an open slot? We've had some back and forth discussion about implementation. We love the idea of adding in these microinteractions but we want to do it systematically. I will get you added to the invite but in the meantime, here is the info to sign up and join:

> Carbon Design Office Hours
> The goal is to help teams adopt Carbon v11, adopt Carbon for IBM Dotcom, achieve cross team alignment, and design in compliance with IBM Design Language.
> 
> Sign-up sheet: https://ibm.box.com/s/hyan6wyze3wgm50rd9yj3rrdv80w4gkt
> 
> https://ibm.webex.com/meet/digdes

**Pull request closed without merging**